### PR TITLE
Remove transaction field from JanusError

### PR DESCRIPTION
**Pull request by vogdb**
Transaction should be only added in proxy error handler (on top of the stack). We currently are not able to pass valid transaction from methods we still want to throw `Janus` errors.

### CODE REVIEW

**Comment by vogdb** [comment]
@tomaszdurka do you mean smth like this?

**Comment by tomaszdurka** [comment]
Yes, but I would remove getWebsocketMessage and only create it when we need it (proxy).

**Comment by vogdb** [comment]
@tomaszdurka please look

File: lib/janus/proxy.js [hunk 1]
```diff
@@ -61,20 +61,24 @@ JanusProxy.prototype.establishConnection = function(fromClientConnection, toJanu
   var connection = proxy.createConnection(fromClientConnection, toJanusConnection);
   proxy.addConnection(connection);
 
-  var handleError = function(error) {
-    proxy.handleError(error, fromClientConnection);
+  var handleError = function(error, request) {
```
Comment (tomaszdurka) [inline lib/janus/proxy.js]: Make transaction arg already.

**Comment by tomaszdurka** [comment]
Search and remove `message['transaction']` from Janus error constructors.

**Comment by vogdb** [comment]
@tomaszdurka please look

File: lib/janus/proxy.js [hunk 2]
```diff
@@ -70,7 +70,9 @@ JanusProxy.prototype.establishConnection = function(fromClientConnection, toJanu
       .then(function(processedRequest) {
         toJanusConnection.send(processedRequest);
       })
-      .catch(handleError);
+      .catch(function(error) {
+        handleError(error, request ? request['transaction'] : null);
```
Comment (tomaszdurka) [inline lib/janus/proxy.js]: Looks good, but you could also pass `request['transaction']` directly.

File: lib/janus/proxy.js [hunk 3]
```diff
@@ -106,15 +110,26 @@ JanusProxy.prototype.establishConnection = function(fromClientConnection, toJanu
...
-    fromClientConnection.send(error.getWebSocketMessage());
+    var errorMessage = {
+      janus: 'error',
+      error: {
+        code: error.code || 490,
+        reason: error.message
+      }
+    };
+    if (transaction) {
+      errorMessage.transaction = transaction;
```
Comment (tomaszdurka) [inline lib/janus/proxy.js]: Should we generate transaction id if not present?

File: lib/janus/session.js [hunk 4]
```diff
@@ -55,7 +55,7 @@ Session.prototype.processMessage = function(message) {
 Session.prototype.onAttach = function(message) {
   var self = this;
   if (!this.pluginRegistry.isAllowedPlugin(message['plugin'])) {
-    return Promise.reject(new JanusError.IllegalPlugin(message['transaction']));
+    return Promise.reject(new JanusError.IllegalPlugin());
```
Comment (tomaszdurka) [inline lib/janus/session.js]: You should probably pass `message['plugin']` here.

**Comment by tomaszdurka** [comment]
Rest looks good.

File: lib/janus/proxy.js (hunk 3 above)
Comment (vogdb) [inline lib/janus/proxy.js]: no. Michel prepared janus.js to accept errors without transaction

File: lib/janus/session.js (hunk 4 above)
Comment (vogdb) [inline lib/janus/session.js]: Well, `JanusError.IllegalPlugin` does not require plugin.

**Comment by vogdb** [comment]
@tomaszdurka please look

**Comment by vogdb** [comment]
@tomaszdurka please review

File: lib/janus/session.js (hunk 4 above)
Comment (tomaszdurka) [inline lib/janus/session.js]: Ah, I looked for InvalidPlugin ...

**Comment by tomaszdurka** [comment]
Looks good.

**Pull request merged**